### PR TITLE
Add a test for SYCL_INTEL_local_memory extension

### PR DESCRIPTION
**Pull request by sergey-semenov**

### CODE REVIEW

**Comment by sergey-semenov** [comment]
Depends on and checks the changes from https://github.com/intel/llvm/pull/3329 and https://github.com/intel/llvm/pull/3228

**Review by Pennycook — DISMISSED** [review]
This looks great.

**Comment by Pennycook** [comment]
@jbrodman suggested that it might be a good idea to demonstrate some alternative ways to use the pointer, in case developers are basing their code on our tests, e.g.:

```c++
auto Ptr = group_local_memory_for_overwrite<int[WgSize]>(Item.get_group());
```
and
```c++
Foo& Ref = *group_local_memory(Item.get_group(), ...);
```

If all of these syntax options are functionally equivalent (as I think they are) then we might not need to actually test them, but it might still be worth adding a comment or two to let readers know other options are available.

**Comment by sergey-semenov** [comment]
@Pennycook Added a comment on this.

**Review by vladimirlaz — APPROVED** [review]

File: SYCL/GroupLocalMemory/group_local_memory.cpp
Comment (vladimirlaz) [inline SYCL/GroupLocalMemory/group_local_memory.cpp:1]: New directory is added. Could you please set correct code owner (in .github/CODEOWNERS)

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/GroupLocalMemory/group_local_memory.cpp
Comment (sergey-semenov) [reply SYCL/GroupLocalMemory/group_local_memory.cpp:1]: Added myself as the code owner. @Pennycook Do you mind being set as the second code owner for the group local memory tests? (I this can be done as a separate PR, so that it does not block merging this one).

Comment (Pennycook) [reply SYCL/GroupLocalMemory/group_local_memory.cpp:1]: > @Pennycook Do you mind being set as the second code owner for the group local memory tests?

Fine by me.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**
